[docs] Add notes for faster CI/CD testing when writing apps (#19287)
yugabyte/yugabyte-db: docs/content/preview/develop/best-practices-ysql.md

## TRUNCATE tables instead of DELETE

[TRUNCATE](../../api/ysql/the-sql-language/statements/ddl_truncate/) deletes the database files that store the table and is much faster than [DELETE](../../api/ysql/the-sql-language/statements/dml_delete/) which inserts a _delete marker_ for each row in transactions that are later removed from storage during compaction runs.

## Settings for ci/cd/integration-tests:
Using YugabyteDB in (ci,cd,automated tests) scenarios we can set certain gflags to increase performance:

1. Point gflag `--fs_data_dirs` & `--fs_wal_dirs` to a ramdisk directory. 

> This will make DML,DDL and create,destroy a cluster faster because data will not be written to disk

Reviewer: delete this line as I have mentioned it as part of my previous suggestion.

> 2. Set gflag `--yb_num_shards_per_tserver=1`.

Reviewer: ```suggestion
- Set the flag `--yb_num_shards_per_tserver=1`. Reducing the number of shards lowers overhead when creating or dropping YCQL tables, and writing or reading small amounts of data.
```

> Reducing the number of shards lowers overhead when creating,dropping YCQL tables and writing,reading small amounts of data

Reviewer: delete this line.

> 3. Use colocated databases in YSQL. 

Reviewer: ```suggestion
- Use colocated databases in YSQL. Colocation lowers overhead when creating or dropping YSQL tables, and writing or reading small amounts of data.
```

Reviewer: We don't need to number them (ordering is not needed as they are all single bullet recommendations and not steps).

> Colocation will lower overhead when creating/dropping YSQL tables and writing,reading small amounts of data 

Reviewer: delete this line

> 4. Set gflag `--replication_factor=1`.

Reviewer: ```suggestion
- Set the flag `--replication_factor=1` for test scenarios, as keeping the data three way replicated (default) is not necessary. Reducing that to 1 reduces space usage and increases performance. 
```

> For these testing scenarios, perhaps the default of keeping the data 3-way replicated is not necessary. Reducing that down to 1 cuts space usage and increases perf.

Reviewer: delete this line

> 5. Use `TRUNCATE table1,table2,table3..tablen;` instead of `CREATE TABLE` and `DROP TABLE` between test cases. 

Reviewer: ```suggestion
- Use `TRUNCATE table1,table2,table3..tablen;` instead of `CREATE TABLE`, and `DROP TABLE` between test cases. 
```
